### PR TITLE
Allow no/empty password (Implements #674)

### PR DESCRIPTION
**Pull request by Empty2k12**
**This allows wallets to have a zero length (`''`) password. (As requested in #674)**

When generating a new wallet and pressing enter both times when you are asked to set a password it will present you with the following warning
```
============= WARNING =============
Using no password is very dangerous
===================================
Abort? [Y/n]
```
To proceed with a empty password you have to confirm with a `n`, pressing enter will stop the generation process.

The wallet checks `''` and if it does not unlock the wallet it will ask the user to input the password.

### CODE REVIEW

**Comment by chris-belcher** [comment]
Thanks for doing this!

Please change the target branch to be joinmarket:develop not joinmarket:master

**Comment by Empty2k12** [comment]
@chris-belcher Sorry, should know by now which branch is the development branch. Changed it!

**Comment by juscamarena** [comment]
tACK

File: joinmarket/wallet.py [hunk 1]
```diff
@@ -177,11 +177,16 @@ def read_wallet_file_data(self, filename, pwd=None):
                 self.index_cache += [[0,0]] * (
                     self.max_mix_depth - len(self.index_cache))
         decrypted = False
+        trieddefault = False
         while not decrypted:
-            if pwd:
-                password = pwd
+            if trieddefault:
+                password = ''
+                trieddefault = True
```
Comment (AdamISZ) [inline joinmarket/wallet.py]: I can't see the logic of this (lines 180-184, plus 204-5); `trieddefault` is set to False before the decryption-attempting loop, and the password is set to '' if `trieddefault` is True; but it is not set to True anywhere except within that condition where it has already evaluated as True. I'm guessing what you intended is, line 182 to read `if not trieddefault`, thus to enforce an attempt to unlock with a null password once only.

Comment (AdamISZ) [inline joinmarket/wallet.py]: Yes, after testing with null and non-null password wallets, the functionality seems to be correct with that single edit.

**Comment by AdamISZ** [comment]
ping @Empty2k12 with that one edit ^ i'm happy to merge this.

**Comment by Empty2k12** [comment]
@AdamISZ I implemented the change. You were right. I'm still wondering how I tested both cases and it worked.

**Comment by AdamISZ** [comment]
Getting failures in test_wallets test script. Sorry I don't have time to figure it out now, if someone else did, that'd be great. You can see the errors in the travis log (it's a way up because of the logfile printouts at the bottom).

**Comment by Empty2k12** [comment]
Thank you for the implementation of the test case @peterzen

**Comment by Empty2k12** [comment]
@peterzen [This](https://travis-ci.org/JoinMarket-Org/joinmarket/builds/208319308) is the log output for the last test run.

**Comment by chris-belcher** [comment]
Added my own commit which fixes a small thing that makes the test now work (although coverage reduced so it says test failed)

Will merge with merge.py soon

**Comment by peterzen** [comment]
Hi Chris - re your message; my address is indeed peter@prioritylane.com.  Thanks!

**Pull request merged**